Admin/CrossTenantMailboxMigrationValidation.ps1

```diff
@@ -0,0 +1,447 @@
+    $TargetObject = Get-TargetMailUser $TargetIdentity -ErrorAction SilentlyContinue
+
+    #Validate if SourceObject is present
+    if ($SourceObject) {
+        #Since SourceObject is valid, validate if TargetObject is present
+        if ($TargetObject) {
+            #Check if source mailbox has aux archives and if so throw error, otherwise continue with the rest of validations
+            Write-Verbose -Message "Checking if SOURCE mailbox has any aux-archives present"
+            if ($SourceObject.MailboxLocations -like '*auxArchive*') {
+                Write-Host ">> Error: The SOURCE malbox has an auxArchive and as of now this object can't be migrated"
+                exit
+            } else {
+                Write-Verbose -Message "No aux archives are present on SOURCE malbox"
+
+                #Verify if SOURCE mailbox is part of the Mail-Enabled Security Group defined on the SOURCE organization relationship
+                Write-Verbose -Message "Informational: Checking if the SOURCE mailbox is a member of the SOURCE organization relationship Mail-Enabled Security Group defined on the MailboxMovePublishedScopes"
+                $SourceTenantOrgRelationship = Get-SourceOrganizationRelationship | Where-Object { ($_.MailboxMoveCapability -eq "RemoteOutbound") -and ($null -ne $_.OauthApplicationId) }
+                if ((Get-SourceDistributionGroupMember $SourceTenantOrgRelationship.MailboxMovePublishedScopes[0]).Name -contains $SourceObject.Name) {
+                    Write-Host ">> SOURCE mailbox is within the MailboxMovePublishedScopes" -ForegroundColor Green
+                } else {
+                    Write-Host ">> Error: SOURCE mailbox is NOT within the MailboxMovePublishedScopes. The migration will fail if you don't correct this" -ForegroundColor Red
+                }
+
+                #Check the recoverableItems quota on TARGET MailUser and compare it with the SOURCE mailbox occupied quota
+                Write-Verbose -Message "Checking if the current dumpster size on SOURCE mailbox is bigger than the TARGET MailUser recoverable items quota"
+                if (((Get-SourceMailboxStatistics $SourceIdentity).TotalDeletedItemSize -replace '^.+\((.+\))', '$1' -replace '\D' -as [int]) -gt ([int64]($TargetObject.RecoverableItemsQuota -replace '^.+\((.+\))', '$1' -replace '\D'))) {
+                    #if (([int64]($SourceObject.RecoverableItemsQuota -replace '^.+\((.+\))','$1' -replace '\D')) -ne ([int64]($TargetObject.RecoverableItemsQuota -replace '^.+\((.+\))','$1' -replace '\D'))){
+                    Write-Host ">> Error: Dumpster size on SOURCE mailbox is bigger than TARGET MailUser RecoverableItemsQuota. This might cause the migration to fail" -ForegroundColor Red
+                    #Write-Host ">> Error: RecoverableItemsQuota on SOURCE mailbox is not the same as TARGET MailUser RecoverableItemsQuota. This will cause the migration to fail" -ForegroundColor Red
+                }
```

Comment:
```suggestion
                if (((Get-SourceMailboxStatistics $SourceIdentity).TotalDeletedItemSize -replace '^.+\((.+\))', '$1' -replace '\D' -as [int]) -gt ([int64]($TargetObject.RecoverableItemsQuota -replace '^.+\((.+\))', '$1' -replace '\D'))) {
                    Write-Host ">> Error: Dumpster size on SOURCE mailbox is bigger than TARGET MailUser RecoverableItemsQuota. This might cause the migration to fail" -ForegroundColor Red
                }
```

Remove commented-out code.